src/runtime/internal/style_manager.ts

```diff
@@ -53,14 +55,19 @@ export function delete_rule(node: Element & ElementCSSInlineStyle, name?: string
 		)
 		.join(', ');
 
-	if (name && !--active) clear_rules();
+	active = Math.max(0, active - 1);
+	if (name && !active) clear_rules();
```

Comment:
I've been working off of this branch (since my Svelte app needs iframe support) and noticed that this change causes a previously working transition of mine to get cut short mid-transition.

It's more noticeable with a longer transition duration. For example, with duration=5s, the first 2s or so of the transition work as expected, but then it jumps to the final position suddenly. This is the case both in and outside of an iframe.

My use case is pretty simple, just: `<div transition:slide={...}>`. Reverting just the above 3 lines fixes the issue for me, although then the leak is still present of course.